### PR TITLE
OADP-741: Add status conditions to VSB & VSR

### PR DESCRIPTION
**Pull request by eemcmullan**
```
apiVersion: datamover.oadp.openshift.io/v1alpha1
kind: VolumeSnapshotBackup
metadata:
  creationTimestamp: "2022-08-11T14:49:27Z"
  generateName: vsb-
  generation: 1
  labels:
    velero.io/backup-name: tb41
  name: vsb-68m2r
  namespace: mssql-persistent
  resourceVersion: "4806532"
  uid: e605b5a5-2b7d-450e-ba33-947ba68629e4
spec:
  protectedNamespace: openshift-adp
  resticSecretRef:
    name: velero-sample-1-volsync-restic
  volumeSnapshotContent:
    name: snapcontent-f43fb30c-2aaa-4366-a0c7-354adc0b0ee9
status:
  conditions:
  - lastTransitionTime: "2022-08-11T14:51:08Z"
    message: Reconcile complete
    reason: Complete
    status: "True"
    type: Reconciled
  phase: InProgress
  resticrepository: s3:s3.amazonaws.com/emcmullavelero-test/openshift-adp/snapcontent-f43fb30c-2aaa-4366-a0c7-354adc0b0ee9-pvc
  sourcePVCData:
    name: mssql-pvc
    size: 10Gi
    storageClassName: gp2-csi
  volumeSnapshotClassName: example-snapclass
```
  
```
apiVersion: datamover.oadp.openshift.io/v1alpha1
kind: VolumeSnapshotRestore
metadata:
  creationTimestamp: "2022-08-11T15:01:03Z"
  generateName: vsr-
  generation: 1
  labels:
    velero.io/persistent-volume-claim-name: mssql-pvc
    velero.io/restore-name: tr42
  name: vsr-sw984
  namespace: mssql-persistent
  resourceVersion: "4832640"
  uid: 2e25f814-a9d3-4322-9d0c-017391afc0da
spec:
  protectedNamespace: openshift-adp
  resticSecretRef:
    name: velero-sample-1-volsync-restic
  volumeSnapshotMoverBackupRef:
    resticrepository: s3:s3.amazonaws.com/emcmullavelero-test/openshift-adp/snapcontent-f43fb30c-2aaa-4366-a0c7-354adc0b0ee9-pvc
    sourcePVCData:
      name: mssql-pvc
      size: 10Gi
      storageClassName: gp2-csi
    volumeSnapshotClassName: example-snapclass
status:
  conditions:
  - lastTransitionTime: "2022-08-11T15:01:03Z"
    message: Reconcile complete
    reason: Complete
    status: "True"
    type: Reconciled
  phase: Completed
  snapshotHandle: snap-009bba11d42eb6694
```

### CODE REVIEW

File: controllers/volumesnapshotbackup_controller.go [hunk 1]
```diff
@@ -164,24 +142,20 @@ func (r *VolumeSnapshotBackupReconciler) Reconcile(ctx context.Context, req ctrl
...
-		r.Log.Error(err, "unable to fetch VolumeSnapshotBackup CR")
-		return result, err
-	}
-
 	statusErr := r.Client.Status().Update(ctx, &vsb)
 	if err == nil { // Don't mask previous error
 		err = statusErr
 	}
 
+	VSBComplete, err := r.setVSBRepSourceStatus(r.Log)
```
Comment (shubham-pampattiwar) [inline controllers/volumesnapshotbackup_controller.go:150]: Shouldnt we handle err != nil condition here ?

Comment (eemcmullan) [reply controllers/volumesnapshotbackup_controller.go:150]: In this case we are handling the err by checking for `!VSBComplete`

Comment (kaovilai) [reply controllers/volumesnapshotbackup_controller.go:150]: So you're saying err is useless. So then why is it even returned?

Comment (eemcmullan) [reply controllers/volumesnapshotbackup_controller.go:150]: Not useless. If there is an error it is also false. However sometimes it is false without being an err.

**Review by shubham-pampattiwar — APPROVED** [review]

**Pull request merged**